Atomic/syscontainers.py

```diff
@@ -1905,14 +1905,14 @@ def mount_from_storage(self, img, destination, upperdir=None, workdir=None, debu
 
     @staticmethod
     def _correct_dir_permissions_for_user(path):
-        os.chmod(path, 0o777)
+        os.chmod(path, 0o700)
         for root, dirs, files in os.walk(path, topdown=False, followlinks=True):
             if os.path.islink(root):
                 continue
             for d in dirs:
                 fullpath = os.path.join(root, d)
                 if not os.path.islink(fullpath):
-                    os.chmod(fullpath, 0o777)
+                    os.chmod(fullpath, 0o700)
             for f in files:
                 fullpath = os.path.join(root, f)
                 if not os.path.islink(fullpath):
```

Comment:
the problem with the directories is that we would not be able to write any file there during the checkout (we would need something like `tar --delay-directory-restore`).  Similarly we would not be able to delete the checkout since we can't delete files that are under a not writeable directory.